packages/Toggle/src/Toggle.js

```diff
@@ -1,7 +1,370 @@
+// which can't use the contained input's focus pseudo-class.
+const focusStateStyle = css`
+  transition: all ${transitionInMS}ms ease-in-out;
+  position: absolute;
+  top: -2px;
+  bottom: -2px;
+  left: -2px;
+  right: -2px;
+  border: 2px solid rgba(67, 177, 229, 0.25);
+  border-radius: 50px;
+  opacity: 0;
+  transform: scale(0.8);
+
+  ${toggleInput.selector}:focus ~ & {
+    opacity: 1;
+    transform: scale(1);
+  }
+`;
+
+const getContainerStyles = ({ size, disabled }) => {
+  const sizeStyle = {
+    default: css`
+      height: 32px;
+      width: 62px;
+    `,
+
+    small: css`
+      height: 22px;
+      width: 40px;
+    `,
+
+    xsmall: css`
+      height: 14px;
+      width: 26px;
+    `,
+  };
+
+  return cx(
+    css`
+      position: relative;
+      display: inline-block;
+      cursor: ${disabled ? 'not-allowed' : 'pointer'};
+    `,
+    sizeStyle[size],
+  );
+};
+
+const getGrooveStyles = ({ variant, checked, disabled }) => {
+  const variants = {
+    default: (() => {
+      if (disabled) {
+        return css`
+          background-color: rgba(29, 36, 36, 0.1);
+          border-color: rgba(18, 22, 22, 0.05);
+        `;
+      }
+
+      const variantStyle = css`
+        box-shadow: inset 0 0 5px rgba(0, 0, 0, 0.1);
+      `;
+
+      if (checked) {
+        return variantStyle;
+      }
+
+      return cx(
+        variantStyle,
+        css`
+          background-color: rgba(29, 36, 36, 0.08);
+          border-color: rgba(0, 0, 0, 0.03);
+        `,
+      );
+    })(),
+
+    dark: (() => {
+      if (disabled) {
+        return css`
+          background-color: rgba(255, 255, 255, 0.15);
+          border-color: rgba(255, 255, 255, 0.1);
+        `;
+      }
+
+      const variantStyle = css`
+        box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.15);
+      `;
+
+      if (checked) {
+        return variantStyle;
+      }
+
+      return cx(
+        variantStyle,
+        css`
+          background-color: rgba(29, 36, 36, 0.6);
+          border-color: rgba(18, 22, 22, 0.1);
+        `,
+      );
+    })(),
+  };
+
+  const baseStyle = css`
+    transition: ${transitionInMS}ms all ease-in-out, 0 background-color linear;
+    display: inline-block;
+    overflow: hidden;
+    flex-shrink: 0;
+    position: relative;
+    border-radius: 50px;
+    overflow: hidden;
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    border: 1px solid;
+
+    // We're animating this pseudo-element in order to give the toggle groove
+    // background an animation in and out.
+    &:before {
+      content: '';
+      transition: ${transitionInMS}ms all ease-in-out;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      border-radius: 50px;
+      background-color: ${colors.mongodb.blue};
+      box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.1);
+      opacity: 0;
+      transform: scale(0.85);
+    }
+  `;
+
+  if (disabled) {
+    return cx(
```

Comment:
What I love about using cx is that it clearly shows which styles will take precedence, as last one in wins.